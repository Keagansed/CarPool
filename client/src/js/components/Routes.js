import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';

class Routes  extends Component {
    render(){
        return(
            <h1>Routes</h1>
        );
    }
}

export default Routes;