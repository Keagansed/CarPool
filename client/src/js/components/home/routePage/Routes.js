import { observer } from 'mobx-react';
import React, { Component } from 'react';

import Route from './Route'
import RouteStore from '../../../stores/RouteStore';

@observer class Routes  extends Component {

    constructor(){
        super()
        // Just for dummy data, remove when necessary
        this.state = {
            daysOne: {
                monday: true,
                tuesday: true,
                wednesday: true,
                thursday: true,
                friday: true,
                saturday: false,
                sunday: false,
            }
        }
    }

    componentWillMount(){
        this.props.store.getRoutes(this.props.token);
    }

    render(){
        // Temporarily commented out to prevent warnings, uncomment when using 'Routes' instead of dummy route
        // const Routes = this.props.store.routes.map(route => 
        //     <Route key={route._id} store={new RouteStore(route.routeName, route.startLocation, route.endLocation, route.days, route.time, route.repeat)}/>
        // )

        return(
            <div>
                {/*What will come here -> {Routes} */}
                
                {/*Just an example... */}
                <Route key={1} store={new RouteStore("Home to Work", "91 Jagluiperd Street", "South Street", this.state.daysOne, "08:00", true)}/>
                <Route key={2} store={new RouteStore("Work to Home", "South Street", "91 Jagluiperd Street", this.state.daysOne, "16:30", true)}/>                                                                
            </div>
        );
    }
}

export default Routes;