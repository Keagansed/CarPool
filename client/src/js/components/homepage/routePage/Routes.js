import { observer } from 'mobx-react';
import React, { Component } from 'react';

import Route from './Route'
import RouteStore from '../../../stores/RouteStore';

@observer class Routes  extends Component {

    componentWillMount(){
        this.props.store.getRoutes(this.props.token);
    }

    

    render(){

        const Routes = this.props.store.routes.map(route => 
            <Route key={route._id} store={new RouteStore(route.routeName, route.startLocation, route.endLocation, route.days, route.time, route.repeat)}/>
        )

        return(
            <div className="list-group bg-info">

                {Routes}
                                                                                
            </div>
        );
    }
}

export default Routes;