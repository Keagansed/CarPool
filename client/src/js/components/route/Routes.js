import { observer } from 'mobx-react';
import React, { Component } from 'react';

import  "../../../css/components/Spinner.css"

import Route from './Route'
import RouteStore from '../../stores/RouteStore';

@observer class Routes  extends Component {

    constructor(){
        super()
        // Just for dummy data, remove when necessary
        this.state = {
            daysOne: {
                monday: true,
                tuesday: true,
                wednesday: true,
                thursday: true,
                friday: true,
                saturday: false,
                sunday: false,
            }
        }
    }

    componentWillMount(){
        this.props.store.getRoutes(this.props.token);
    }

    renderRoutes = () => {
        const Routes = this.props.store.routes.map(route =>             
            <Route key={route._id} store={new RouteStore(route.routeName, route.startLocation, route.endLocation, route.days, route.time, route.repeat, route._id)}/>
        )
        
        if(Routes.length > 0) {
            return Routes;
        }else {
            return(
                <h5 className="txt-center mtop-10px txt-white">
                    No Routes
                </h5>
            );
        }
    }

    renderLoading = () => {
        return(
            <div>
                <div className="spinner">
                    <div className="double-bounce1"></div>
                    <div className="double-bounce2"></div>
                </div>
            </div>
        )
    }

    render(){
        
        if(this.props.store.loadingRoutes) {
            return(
                <div className="scroll-vert">
                    {this.renderLoading()}
                </div>
            );
        }
        else{
            return(
                <div className="scroll-vert">
                    {this.renderRoutes()}
                </div>
            );
        }
    }
}

export default Routes;