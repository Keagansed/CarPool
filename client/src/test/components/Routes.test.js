import React from 'react';
import Routes from '../../js/components/route/Routes';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

class MockStore {
    loadingRoutes = false;
    routes = [];

    getRoutes = (token) => {
        let route = {
            _id: "123",
            routeName: "TestRoute",
            startLocation: {},
            endLocation: {},
            days: "",
            time: "00:00",
            repeat: false,
        }
        this.routes.push(route);
    }
}

describe('Routes Component', () => {
    let token, container, instance, mockStore;

    beforeEach(() => {
        mockStore = new MockStore();
        token = "testtoken123";
        container = shallow(<Routes store={mockStore} token={token} />);
        instance = container.instance();
    });

    it ('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><Routes store={mockStore} token={token} /></MemoryRouter>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it ('renders correctly', () => {
        expect(container.length).toEqual(1);
    });

    it ('loads spinner when loading routes', () => {
        mockStore = new MockStore();
        mockStore.loadingRoutes = true;
        container = shallow(<Routes store={mockStore} token={token} />);
        expect(container.exists(".spinner")).toEqual(true);
        expect(container.exists("Route")).toEqual(false);
    });

    it ('displays route component after loading', () => {
        expect(container.exists("Route")).toEqual(true);
        expect(container.exists(".spinner")).toEqual(false);
    });

});