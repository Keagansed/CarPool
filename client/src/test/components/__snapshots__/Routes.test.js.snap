// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component captures snapshot 1`] = `
<div
  className="scroll-vert"
>
  <div
    className="container-fluid bg-purple bordbot-2px-white"
  >
    <a
      href="/HomePage/RouteMatches/123"
      onClick={[Function]}
    >
      <div
        className="row txt-white padver-10px"
      >
        <div
          className="col-9"
        >
          <div
            className="col-12"
          >
            <h5>
              TestRoute
            </h5>
          </div>
          <div
            className="col-12"
          >
            Mon Tue Wed Thu Fri
          </div>
        </div>
        <div
          className="col-3 vertical-right"
        >
          <div
            className="col-12"
          >
            <h5>
              <i
                className="fa fa-chevron-circle-right"
              />
            </h5>
          </div>
          <div
            className="col-12"
          >
            00:00
          </div>
        </div>
      </div>
    </a>
  </div>
  <div
    className="container-fluid bg-purple bordbot-2px-white"
  >
    <a
      href="/HomePage/RouteMatches/123"
      onClick={[Function]}
    >
      <div
        className="row txt-white padver-10px"
      >
        <div
          className="col-9"
        >
          <div
            className="col-12"
          >
            <h5>
              TestRoute
            </h5>
          </div>
          <div
            className="col-12"
          >
            Mon Tue Wed Thu Fri
          </div>
        </div>
        <div
          className="col-3 vertical-right"
        >
          <div
            className="col-12"
          >
            <h5>
              <i
                className="fa fa-chevron-circle-right"
              />
            </h5>
          </div>
          <div
            className="col-12"
          >
            00:00
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
